src/algorithms/tracking/TrackerHitReconstruction.cc

```diff
@@ -85,6 +89,74 @@ std::unique_ptr<edm4eic::TrackerHitCollection> TrackerHitReconstruction::process
 
     }
 
+    //------------------------------------------------
+    // Example code of adding random noise hits
+    // Shujie Li, 08.2024
+    // use the first raw hits to obtain the volume ID of the detector system
+    // this also make sure the hit container has at least one hit
+    int num_hits=10; // total number of noise hits in this detector volume. Should move this to the config file
+    std::vector<uint64_t> noise_ids;
+
+    // generate valid random cell id
+    for (const auto& hit0 : raw_hits) {
+        uint64_t id0,vol_id;
+        id0 = hit0.getCellID();
+        // vol ID is the last 8 bits in Si tracker. Need to make it more flexible
+        // may want to predefine the layer/module ID as well to speed up the radom ID generation
+        vol_id = id0 & 0xFF;
```

Comment:
Should we not call it "system ID"?
```suggestion
        uint64_t id0, system_id;
        id0 = hit0.getCellID();
        // system ID is the last 8 bits in Si tracker. Need to make it more flexible
        // may want to predefine the layer/module ID as well to speed up the radom ID generation
        system_id = id0 & 0xFF;
```